src/debug/telemetry/telemetry.c

```diff
@@ -205,6 +205,67 @@ int free_performance_data(struct perf_data_item_comp *item)
 	return IPC4_SUCCESS;
 }
 
+int get_extended_performance_data(struct extended_global_perf_data * const ext_global_perf_data)
+{
+	if (!ext_global_perf_data)
+		return -EINVAL;
+
+	size_t slots_count;
+	size_t slot_idx = 0;
+	uint64_t total_dsp_cycles[CONFIG_MAX_CORE_COUNT];
+
+	/* TODO
+	 * Seting temporary values here.
+	 * Replace this with actual total dsp cycles info once it is available.
+	 */
+	for (size_t core_id = 0; core_id < CONFIG_MAX_CORE_COUNT; ++core_id)
+		total_dsp_cycles[core_id] = 1;
+
+	for (size_t core_id = 0; core_id < CONFIG_MAX_CORE_COUNT; ++core_id) {
+		if (!(cpu_enabled_cores() & BIT(core_id)))
+			continue;
+
+		memset(&ext_global_perf_data->perf_items[slot_idx], 0,
+		       sizeof(struct ext_perf_data_item));
+		/* 0 is id of base_fw */
```

Comment:
0 is the module ID of the base firmware, right? That's what you mean? Then what does the core ID have to do with module IDs?

Reply:
In reference FW it was basefw_id + core_id, which effectively is just core id. I also have no idea why it was written like this. This record contains total cycles used by specific core. I'll just remove this comment.